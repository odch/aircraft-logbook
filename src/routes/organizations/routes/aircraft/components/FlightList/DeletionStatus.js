import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { FormattedMessage } from 'react-intl'
import { withStyles } from '@material-ui/core'
import Typography from '@material-ui/core/Typography'
import { flight as flightShape } from '../../../../../../shapes'

const styles = theme => ({
  status: {
    textTransform: 'uppercase',
    backgroundColor: theme.palette.grey[200],
    borderRadius: theme.typography.pxToRem(5),
    padding: theme.typography.pxToRem(5),
    display: 'inline-block',
    verticalAlign: 'top',
    fontSize: '0.8em'
  },
  deleted: {
    backgroundColor: theme.palette.error.dark,
    color: '#fff'
  },
  replaced: {
    backgroundColor: '#e5be73',
    color: '#000'
  }
})

const DeletionStatus = ({ flight, classes, className }) => (
  <Typography
    component="span"
    className={classNames(
      classes.status,
      flight.replacedWith ? classes.replaced : classes.deleted,
      className
    )}
  >
    <FormattedMessage
      id={`flightlist.flight.${flight.replacedWith ? 'replaced' : 'deleted'}`}
    />
  </Typography>
)

DeletionStatus.propTypes = {
  flight: flightShape.isRequired,
  classes: PropTypes.object.isRequired,
  className: PropTypes.string
}

export default withStyles(styles)(DeletionStatus)
