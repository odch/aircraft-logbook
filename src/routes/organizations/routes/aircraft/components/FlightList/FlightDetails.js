import React from 'react'
import { useIntl } from 'react-intl'
import TextField from '@material-ui/core/TextField'
import Grid from '@material-ui/core/Grid'
import Divider from '@material-ui/core/Divider'
import { aircraft, flight } from '../../../../../../shapes'
import {
  formatDate,
  formatTime,
  getTimeDiff
} from '../../../../../../util/dates'

const FlightDetails = ({ aircraft, flight }) => {
  const intl = useIntl()
  return (
    <div>
      <Grid container>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'date',
              formatDate(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField('pilot', getMemberName(flight.pilot), intl)}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField('instructor', getMemberName(flight.instructor), intl)}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'departureaerodrome',
              getAerodromeName(flight.departureAerodrome),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'blockofftime',
              formatTimeWithUtc(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'takeofftime',
              formatTimeWithUtc(
                flight.takeOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'destinationaerodrome',
              getAerodromeName(flight.destinationAerodrome),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'blockontime',
              formatTimeWithUtc(
                flight.blockOnTime,
                flight.destinationAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'landingtime',
              formatTimeWithUtc(
                flight.landingTime,
                flight.destinationAerodrome.timezone
              ),
              intl
            )}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField('nature', getFlightNature(flight.nature, intl), intl)}
          </Grid>
          <Grid item xs={12} sm={4}>
            {renderField(
              'fueluplift',
              getFuel(
                flight.fuelUplift,
                flight.fuelType,
                aircraft.settings.fuelTypes
              ),
              intl
            )}
          </Grid>
          <Grid item xs={12} sm={4}>
            <Grid item xs={12} sm={4}>
              {renderField('oiluplift', getOil(flight.oilUplift), intl)}
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField('landings', flight.landings, intl)}
          </Grid>
          <Grid item xs={12} sm={4}>
            {renderField('personsonboard', flight.personsOnBoard || '-', intl)}
          </Grid>
          <Grid item xs={12} sm={8}>
            {renderField('remarks', flight.remarks || '-', intl, true)}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={6} sm={4}>
            {renderField(
              'flighthours_time',
              getTimeDiff(flight.takeOffTime, flight.landingTime),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'blockhours',
              getTimeDiff(flight.blockOffTime, flight.blockOnTime),
              intl
            )}
          </Grid>
        </Grid>
        {flight.counters && (
          <Grid item xs={12} container>
            {flight.counters.flightTimeCounter && (
              <Grid item xs={6} sm={4}>
                {renderField(
                  'flighthours_counter',
                  getDecimalRange(flight.counters.flightTimeCounter),
                  intl
                )}
              </Grid>
            )}
            {flight.counters.engineTimeCounter && (
              <Grid item xs={6} sm={4}>
                {aircraft.settings.engineHoursCounterEnabled &&
                  renderField(
                    'enginehours',
                    getDecimalRange(flight.counters.engineTimeCounter),
                    intl
                  )}
              </Grid>
            )}
          </Grid>
        )}
      </Grid>
      <Divider />
      <Grid item xs={12} container>
        <Grid item xs={6} sm={4}>
          {renderField(
            'total.flighthours',
            format2Decimals(flight.counters.flightHours.end),
            intl
          )}
        </Grid>
        {flight.counters.engineHours && (
          <Grid item xs={6} sm={4}>
            {renderField(
              'total.enginehours',
              format2Decimals(flight.counters.engineHours.end),
              intl
            )}
          </Grid>
        )}
        <Grid item xs={6} sm={4}>
          {renderField('total.landings', flight.counters.landings.end, intl)}
        </Grid>
      </Grid>
    </div>
  )
}

const renderField = (label, value, intl, multiline = false) => (
  <TextField
    label={intl.formatMessage({ id: `flightlist.${label}` })}
    value={value}
    margin="normal"
    InputProps={{
      readOnly: true,
      disableUnderline: true
    }}
    fullWidth
    multiline={multiline}
  />
)

const getMemberName = member =>
  member ? `${member.firstname} ${member.lastname}` : '-'

const getAerodromeName = aerodrome =>
  `${aerodrome.identification} (${aerodrome.name})`

const getFlightNature = (nature, intl) =>
  intl.formatMessage({ id: `flight.nature.${nature}` })

const getFuel = (fuelUplift, fuelType, aircraftFuelTypes) =>
  fuelUplift && fuelUplift > 0
    ? `${fuelUplift}L (${getFuelTypeDescription(fuelType, aircraftFuelTypes)})`
    : '-'

const getFuelTypeDescription = (fuelType, aircraftFuelTypes) => {
  const foundType = aircraftFuelTypes.find(type => type.name === fuelType)
  if (foundType && foundType.description) {
    return foundType.description
  }
  return fuelType
}

const getOil = oilUplift => (oilUplift && oilUplift > 0 ? `${oilUplift}L` : '-')

const getDecimalRange = range => {
  const diff = range.end - range.start

  const startFormatted = format2Decimals(range.start)
  const endFormatted = format2Decimals(range.end)
  const diffFormatted = format2Decimals(diff)

  return `${diffFormatted} (${startFormatted} - ${endFormatted})`
}

const format2Decimals = value => Number(value / 100).toFixed(2)

const formatTimeWithUtc = (timestamp, timezone) =>
  `${formatTime(timestamp, timezone)} (${formatTime(timestamp, 'UTC')} UTC)`

FlightDetails.propTypes = {
  aircraft,
  flight
}

export default FlightDetails
