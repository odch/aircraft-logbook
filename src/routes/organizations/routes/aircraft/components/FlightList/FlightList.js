import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import withStyles from '@material-ui/core/styles/withStyles'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import {
  aircraft as aircraftShape,
  flight as flightShape,
  organization as organizationShape
} from '../../../../../../shapes'
import { formatDate, formatTime } from '../../../../../../util/dates'

const styles = theme => ({
  table: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing.unit * 3 * 2)]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  hiddenBelow800: {
    [theme.breakpoints.down(800 + theme.spacing.unit * 3 * 2)]: {
      display: 'none'
    }
  },
  hiddenBelow600: {
    [theme.breakpoints.down(600 + theme.spacing.unit * 3 * 2)]: {
      display: 'none'
    }
  }
})

class FlightList extends React.Component {
  render() {
    const { flights, classes } = this.props
    return (
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>
              <FormattedMessage id="flightlist.date" />
            </TableCell>
            <TableCell>
              <FormattedMessage id="flightlist.pilot" />
            </TableCell>
            <TableCell className={classes.hiddenBelow600}>
              <FormattedMessage id="flightlist.departureaerodrome" />
            </TableCell>
            <TableCell>
              <FormattedMessage id="flightlist.destinationaerodrome" />
            </TableCell>
            <TableCell className={classes.hiddenBelow800}>
              <FormattedMessage id="flightlist.blockofftime" />
            </TableCell>
            <TableCell className={classes.hiddenBelow800}>
              <FormattedMessage id="flightlist.blockontime" />
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {flights.map(flight => {
            return (
              <TableRow key={flight.id}>
                <TableCell>{formatDate(flight.blockOffTime)}</TableCell>
                <TableCell>
                  {`${flight.member.firstname} ${flight.member.lastname}`}
                </TableCell>
                <TableCell className={classes.hiddenBelow600}>
                  {flight.departureAerodrome.name}
                </TableCell>
                <TableCell>{flight.destinationAerodrome.name}</TableCell>
                <TableCell className={classes.hiddenBelow800}>
                  {formatTime(flight.blockOffTime)}
                </TableCell>
                <TableCell className={classes.hiddenBelow800}>
                  {formatTime(flight.blockOnTime)}
                </TableCell>
              </TableRow>
            )
          })}
        </TableBody>
      </Table>
    )
  }
}

FlightList.propTypes = {
  organization: organizationShape,
  aircraft: aircraftShape,
  flights: PropTypes.arrayOf(flightShape),
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(FlightList)
