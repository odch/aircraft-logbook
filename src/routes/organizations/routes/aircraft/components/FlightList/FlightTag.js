import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { withStyles } from '@material-ui/core'
import Typography from '@material-ui/core/Typography'

const styles = theme => ({
  tag: {
    textTransform: 'uppercase',
    backgroundColor: theme.palette.grey[200],
    borderRadius: theme.typography.pxToRem(5),
    padding: theme.typography.pxToRem(5),
    display: 'inline-block',
    verticalAlign: 'top',
    fontSize: '0.8em'
  }
})

const FlightTag = ({ children, classes, className }) => (
  <Typography component="span" className={classNames(classes.tag, className)}>
    {children}
  </Typography>
)

FlightTag.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  className: PropTypes.string
}

export default withStyles(styles)(FlightTag)
