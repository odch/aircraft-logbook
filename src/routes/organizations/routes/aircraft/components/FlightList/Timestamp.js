import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { withStyles } from '@material-ui/core'
import Typography from '@material-ui/core/Typography'
import { formatDate, formatTime } from '../../../../../../util/dates'

const styles = {
  timestamp: {
    fontStyle: 'italic',
    fontSize: '0.8em'
  }
}

const Timestamp = ({
  operation,
  timestamp,
  member: { firstname, lastname },
  classes
}) => (
  <Typography color="textSecondary" className={classes.timestamp}>
    <FormattedMessage
      id={`flightlist.flight.timestamp.${operation}`}
      values={{
        timestamp: `${formatDate(timestamp)} ${formatTime(timestamp)}`,
        member: `${firstname} ${lastname}`
      }}
    />
  </Typography>
)

Timestamp.propTypes = {
  operation: PropTypes.oneOf(['created', 'deleted', 'replaced']).isRequired,
  timestamp: PropTypes.object.isRequired,
  member: PropTypes.shape({
    firstname: PropTypes.string.isRequired,
    lastname: PropTypes.string.isRequired
  }).isRequired,
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(Timestamp)
