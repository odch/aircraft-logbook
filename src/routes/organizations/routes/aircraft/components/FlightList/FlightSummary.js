import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import { formatDate, formatTime } from '../../../../../../util/dates'
import { flight as flightShape } from '../../../../../../shapes'
import DeletionStatus from './DeletionStatus'
import Version from './Version'
import FlightTag from './FlightTag'

const styles = theme => ({
  flightHeading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '100%',
    flexShrink: 0,
    [theme.breakpoints.up(700 + theme.spacing(3 * 2))]: {
      flexBasis: '50%'
    }
  },
  flightSecondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary
  },
  bold: {
    fontWeight: 'bold'
  },
  deletionStatus: {
    marginRight: '0.5em'
  }
})

const FlightSummary = ({ flight, showDeleted, classes }) => (
  <>
    <div className={classes.flightHeading}>
      <Typography>
        {flight.correction === true ? (
          <FormattedMessage
            id="flightlist.correctionflight.heading"
            values={{
              firstname: flight.pilot.firstname,
              lastname: flight.pilot.lastname
            }}
          />
        ) : flight.version === 0 ? (
          <FormattedMessage
            id="flightlist.flight.heading.draft"
            values={{
              departureAerodrome: (
                <span className={classes.bold}>
                  {flight.departureAerodrome.identification}
                </span>
              ),
              firstname: flight.pilot.firstname,
              lastname: flight.pilot.lastname
            }}
          />
        ) : (
          <FormattedMessage
            id="flightlist.flight.heading"
            values={{
              departureAerodrome: (
                <span className={classes.bold}>
                  {flight.departureAerodrome.identification}
                </span>
              ),
              destinationAerodrome: (
                <span className={classes.bold}>
                  {flight.destinationAerodrome.identification}
                </span>
              ),
              firstname: flight.pilot.firstname,
              lastname: flight.pilot.lastname
            }}
          />
        )}
      </Typography>
      {flight.deleted === true && (
        <DeletionStatus flight={flight} className={classes.deletionStatus} />
      )}
      {flight.version === 0 && (
        <FlightTag>
          <FormattedMessage id="flightlist.flight.version.preflight" />
        </FlightTag>
      )}
      {showDeleted &&
        flight.version > 0 &&
        (flight.replacedWith || flight.replaces) && <Version flight={flight} />}
    </div>
    {flight.correction === true ? (
      <Typography className={classes.flightSecondaryHeading}>
        {formatDate(flight.blockOffTime, flight.departureAerodrome.timezone)},{' '}
        {formatTime(flight.blockOffTime, flight.departureAerodrome.timezone)}
      </Typography>
    ) : flight.version === 0 ? (
      <Typography className={classes.flightSecondaryHeading}>
        {formatDate(flight.blockOffTime, flight.departureAerodrome.timezone)}
      </Typography>
    ) : (
      <Typography className={classes.flightSecondaryHeading}>
        {formatDate(flight.blockOffTime, flight.departureAerodrome.timezone)},{' '}
        {formatTime(flight.blockOffTime, flight.departureAerodrome.timezone)}-
        {formatTime(flight.blockOnTime, flight.destinationAerodrome.timezone)}
      </Typography>
    )}
  </>
)

FlightSummary.propTypes = {
  flight: flightShape.isRequired,
  showDeleted: PropTypes.bool,
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(FlightSummary)
