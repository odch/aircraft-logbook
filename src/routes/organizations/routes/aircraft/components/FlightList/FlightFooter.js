import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl } from 'react-intl'
import withStyles from '@material-ui/core/styles/withStyles'
import Divider from '@material-ui/core/Divider'
import Grid from '@material-ui/core/Grid'
import {
  aircraft as aircraftShape,
  flight as flightShape,
  intl as intlShape
} from '../../../../../../shapes'
import { renderField, formatDecimalNumber } from '../../util/formatUtils'
import Timestamp from './Timestamp'

const styles = {
  dividerDraftFlight: {
    margin: '1em 0'
  }
}

const FlightFooter = ({ flight, aircraft, intl, classes }) => (
  <>
    <Divider
      className={flight.version === 0 ? classes.dividerDraftFlight : null}
    />
    {flight.version > 0 && (
      <Grid item xs={12} container>
        <Grid item xs={6} sm={4}>
          {renderField(
            'total.flighthours',
            formatDecimalNumber(flight.counters.flightHours.end),
            intl
          )}
        </Grid>
        {flight.counters.engineHours && (
          <Grid item xs={6} sm={4}>
            {renderField(
              'total.enginehours',
              formatDecimalNumber(
                flight.counters.engineHours.end,
                aircraft.settings.engineHoursCounterFractionDigits
              ),
              intl
            )}
          </Grid>
        )}
        <Grid item xs={6} sm={4}>
          {renderField('total.landings', flight.counters.landings.end, intl)}
        </Grid>
      </Grid>
    )}
    {flight.createTimestamp && (
      <Grid item xs={12} container>
        <Grid item xs={12}>
          <Timestamp
            operation="created"
            timestamp={flight.createTimestamp}
            member={flight.owner}
          />
          {flight.deleteTimestamp && (
            <Timestamp
              operation={flight.replacedWith ? 'replaced' : 'deleted'}
              timestamp={flight.deleteTimestamp}
              member={flight.deletedBy}
            />
          )}
        </Grid>
      </Grid>
    )}
  </>
)

FlightFooter.propTypes = {
  aircraft: aircraftShape.isRequired,
  flight: flightShape.isRequired,
  intl: intlShape.isRequired,
  classes: PropTypes.object.isRequired
}

export default injectIntl(withStyles(styles)(FlightFooter))
