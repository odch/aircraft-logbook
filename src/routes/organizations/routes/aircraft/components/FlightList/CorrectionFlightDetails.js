import React from 'react'
import { useIntl } from 'react-intl'
import Grid from '@material-ui/core/Grid'
import { aircraft, flight } from '../../../../../../shapes'
import { formatDate } from '../../../../../../util/dates'
import {
  renderField,
  formatTimeWithUtc,
  getMemberName,
  getAerodromeName,
  formatCorrection,
  renderCorrectionSets
} from '../../util/formatUtils'
import FlightFooter from './FlightFooter'

const CorrectionFlightDetails = ({ aircraft, flight }) => {
  const intl = useIntl()
  return (
    <div>
      <Grid container>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'date',
              formatDate(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'time',
              formatTimeWithUtc(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField('pilot', getMemberName(flight.pilot), intl)}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12}>
            {flight.remarks &&
              renderField(
                'correctionremarks',
                flight.remarks || '-',
                intl,
                true
              )}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          {flight.departureAerodrome.identification !==
            flight.destinationAerodrome.identification && (
            <>
              <Grid item xs={12}>
                {renderField(
                  'aerodrome',
                  formatCorrection(
                    getAerodromeName(flight.departureAerodrome),
                    getAerodromeName(flight.destinationAerodrome)
                  ),
                  intl
                )}
              </Grid>
            </>
          )}
        </Grid>
        {renderCorrectionSets(
          flight.counters,
          [
            {
              name: 'flightHours',
              label: 'flighthours',
              type: 'decimal'
            },
            {
              name: 'flightTimeCounter',
              label: 'flighthours_counter',
              type: 'decimal'
            },
            { name: 'landings', label: 'landings' },
            {
              name: 'engineHours',
              label: 'enginehours',
              type: 'decimal',
              fractionDigits: aircraft.settings.engineHoursCounterFractionDigits
            },
            {
              name: 'engineTimeCounter',
              label: 'enginetimecounter',
              type: 'decimal',
              fractionDigits: aircraft.settings.engineHoursCounterFractionDigits
            },
            {
              name: 'engineTachHours',
              label: 'enginetachhours',
              type: 'decimal',
              fractionDigits:
                aircraft.settings.engineTachHoursCounterFractionDigits
            },
            {
              name: 'engineTachCounter',
              label: 'enginetachcounter',
              type: 'decimal',
              fractionDigits:
                aircraft.settings.engineTachHoursCounterFractionDigits
            }
          ],
          intl
        )}
      </Grid>
      <FlightFooter flight={flight} aircraft={aircraft} />
    </div>
  )
}

CorrectionFlightDetails.propTypes = {
  aircraft,
  flight
}

export default CorrectionFlightDetails
