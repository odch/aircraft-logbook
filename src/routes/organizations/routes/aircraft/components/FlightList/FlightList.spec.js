import React from 'react'
import renderIntl from '../../../../../../testutil/renderIntl'
import FlightList from './FlightList'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('FlightList', () => {
            it('renders correctly', () => {
              const flights = [
                {
                  id: 'sStfyLd2XArT7oUZPFDn',
                  departureAerodrome: { name: 'Lommis' },
                  destinationAerodrome: { name: 'Lommis' },
                  blockOffTime: {
                    toDate: () => Date.parse('2018-11-20 10:00')
                  },
                  blockOnTime: {
                    toDate: () => Date.parse('2018-11-20 11:10')
                  },
                  takeOffTime: {
                    toDate: () => Date.parse('2018-11-20 10:10')
                  },
                  landingTime: {
                    toDate: () => Date.parse('2018-11-20 11:00')
                  },
                  pilot: {
                    firstname: 'Max',
                    lastname: 'Muster'
                  }
                },
                {
                  id: 'sStfyLd2XArT7oUZPFDa',
                  departureAerodrome: { name: 'Lommis' },
                  destinationAerodrome: { name: 'Lommis' },
                  blockOffTime: {
                    toDate: () => Date.parse('2018-11-21 10:00')
                  },
                  blockOnTime: {
                    toDate: () => Date.parse('2018-11-21 11:10')
                  },
                  takeOffTime: {
                    toDate: () => Date.parse('2018-11-20 10:10')
                  },
                  landingTime: {
                    toDate: () => Date.parse('2018-11-20 11:00')
                  },
                  pilot: {
                    firstname: 'Hans',
                    lastname: 'Meier'
                  }
                }
              ]

              const tree = renderIntl(
                <FlightList
                  flights={flights}
                  flightDeleteDialog={{ open: false }}
                  pagination={{
                    rowsCount: 25,
                    page: 1,
                    rowsPerPage: 10
                  }}
                  openFlightDeleteDialog={() => {}}
                  closeFlightDeleteDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
