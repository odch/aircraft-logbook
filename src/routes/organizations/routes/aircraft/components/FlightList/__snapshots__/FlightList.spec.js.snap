// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components FlightList renders correctly 1`] = `
<table
  className="MuiTable-root-4 FlightList-table-1"
>
  <thead
    className="MuiTableHead-root-5"
  >
    <tr
      className="MuiTableRow-root-6 MuiTableRow-head-9"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-head-12"
        scope="col"
      >
        <span>
          Datum
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-head-12"
        scope="col"
      >
        <span>
          Pilot
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-head-12 FlightList-hiddenBelow600-3"
        scope="col"
      >
        <span>
          Startflugplatz
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-head-12"
        scope="col"
      >
        <span>
          Zielflugplatz
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-head-12 FlightList-hiddenBelow800-2"
        scope="col"
      >
        <span>
          Block off
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-head-12 FlightList-hiddenBelow800-2"
        scope="col"
      >
        <span>
          Block on
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-19"
  >
    <tr
      className="MuiTableRow-root-6"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13"
      >
        20.11.2018
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13"
      >
        Max Muster
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13 FlightList-hiddenBelow600-3"
      >
        Lommis
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13"
      >
        Lommis
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13 FlightList-hiddenBelow800-2"
      >
        10:00
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13 FlightList-hiddenBelow800-2"
      >
        11:10
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-6"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13"
      >
        21.11.2018
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13"
      >
        Hans Meier
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13 FlightList-hiddenBelow600-3"
      >
        Lommis
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13"
      >
        Lommis
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13 FlightList-hiddenBelow800-2"
      >
        10:00
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-11 MuiTableCell-body-13 FlightList-hiddenBelow800-2"
      >
        11:10
      </td>
    </tr>
  </tbody>
</table>
`;
